odoo/addons/test_rpc/tests/test_error.py

```diff
@@ -65,3 +65,11 @@ def test_02_delete(self):
             )
             self.assertIn("Model: Model A (test_rpc.model_a)", e.faultString)
             self.assertIn("Constraint: test_rpc_model_a_field_b2_fkey", e.faultString)
+
+    def test_03_sql_constraint(self):
+        try:
+            with mute_logger("odoo.sql_db"):
```

Comment:
You can use AssertRaises instead (or assertRaisesRegex to match the message), it is much more cleaner. you can find example in the code.